.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,34 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.234.0/containers/debian
+{
+	"name": "Debian",
+	"build": {
+		"dockerfile": "Dockerfile",
+		// Update 'VARIANT' to pick an Debian version: bullseye, buster
+		// Use bullseye on local arm64/Apple Silicon.
+		"args": { "VARIANT": "buster" }
+	},
+
+	"settings": {},
+
+	"extensions": [],
+
+	"mounts": [ "source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind" ],
+
+	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
+
+	"remoteUser": "vscode",
+	"features": {
```

Comment:
Is there anyway to make this faster by removing some of these?

We don't need to use `k3s`, `minikube`, would be fine. Do we need `docker-in-docker`? I'm pretty sure we don't need Python.

Reply:
- removed `kubectl-helm-minikube` feature, change to install `kubectl` in `DOCKERFILE`.
- removed `git` feature, change to install `git` in `DOCKERFILE`
- leaving the rest of the features, considering they can be quite architecture-dependent and can become quite complicated to write in DOCKERFILE, better rely on the [vscode maintainence](https://github.com/microsoft/vscode-dev-containers/tree/main/script-library)?
- we need `docker-in-docker`, since k3s cluster container is running from the dev-container, this is a `docker-in-docker` setup.
- we need `python` for the `mkdocs`, this gets run as part of the make pre-commit
- Total dev-container build time is about 7 min on 16vcore i7-11800h windows